tests/zkevm/test_worst_stateful_opcodes.py

```diff
@@ -0,0 +1,491 @@
+"""
+
+import pytest
+
+from ethereum_test_forks import Fork
+from ethereum_test_tools import (
+    Account,
+    Address,
+    Alloc,
+    Block,
+    BlockchainTestFiller,
+    Bytecode,
+    Environment,
+    Transaction,
+    While,
+    compute_create_address,
+)
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+REFERENCE_SPEC_GIT_PATH = "TODO"
+REFERENCE_SPEC_VERSION = "TODO"
+
+MAX_CODE_SIZE = 24 * 1024
+
+
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.parametrize(
+    "opcode",
+    [
+        Op.BALANCE,
+    ],
+)
+@pytest.mark.parametrize(
+    "absent_accounts",
+    [
+        True,
+        False,
+    ],
+)
+@pytest.mark.slow()
+def test_worst_address_state_cold(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    fork: Fork,
+    opcode: Op,
+    absent_accounts: bool,
+):
+    """Test running a block with as many stateful opcodes accessing cold accounts."""
+    env = Environment(gas_limit=100_000_000_000)
+    attack_gas_limit = Environment().gas_limit
+
+    gas_costs = fork.gas_costs()
+    intrinsic_gas_cost_calc = fork.transaction_intrinsic_cost_calculator()
+    # For calculation robustness, the calculation below ignores "glue" opcodes like  PUSH and POP.
+    # It should be considered a worst-case number of accounts, and a few of them might not be
+    # targeted before the attacking transaction runs out of gas.
+    num_target_accounts = (
+        attack_gas_limit - intrinsic_gas_cost_calc()
+    ) // gas_costs.G_COLD_ACCOUNT_ACCESS
+
+    blocks = []
+    post = {}
+
+    # Setup
+    # The target addresses are going to be constructed (in the case of absent=False) and called
+    # as addr_offset + i, where i is the index of the account. This is to avoid
+    # collisions with the addresses indirectly created by the testing framework.
+    addr_offset = 100_000
+
+    if not absent_accounts:
+        factory_code = Op.PUSH4(num_target_accounts) + While(
+            body=Op.POP(Op.CALL(address=Op.ADD(addr_offset, Op.DUP6), value=10)),
+            condition=Op.PUSH1(1) + Op.SWAP1 + Op.SUB + Op.DUP1 + Op.ISZERO + Op.ISZERO,
+        )
+        factory_address = pre.deploy_contract(code=factory_code, balance=10**18)
+
+        setup_tx = Transaction(
+            to=factory_address,
+            gas_limit=env.gas_limit,
+            gas_price=10,
+            sender=pre.fund_eoa(),
+        )
+        blocks.append(Block(txs=[setup_tx]))
+
+        for i in range(num_target_accounts):
+            addr = Address(i + addr_offset + 1)
+            post[addr] = Account(balance=10)
+
+    # Execution
+    op_code = Op.PUSH4(num_target_accounts) + While(
+        body=Op.POP(opcode(Op.ADD(addr_offset, Op.DUP1))),
+        condition=Op.PUSH1(1) + Op.SWAP1 + Op.SUB + Op.DUP1 + Op.ISZERO + Op.ISZERO,
+    )
+    op_address = pre.deploy_contract(code=op_code)
+    op_tx = Transaction(
+        to=op_address,
+        gas_limit=attack_gas_limit,
+        gas_price=10,
```

Comment:
```suggestion
```
Can be removed. Same for all txs in the rest of the file.

Reply:
Will do!